drv/lpc55-swd/build.rs

```diff
@@ -36,12 +38,41 @@ fn generate_swd_functions(config: &TaskConfig) -> Result<()> {
     let dest_path = out_dir.join("swd.rs");
     let mut file = std::fs::File::create(dest_path)?;
 
-    let out_cfg = &config.out_cfg;
-    let in_cfg = &config.in_cfg;
     let spi_periph = quote::format_ident!("Fc{}", config.spi_num);
     let flexcomm = quote::format_ident!("FLEXCOMM{}", config.spi_num);
     let spi_regs = quote::format_ident!("SPI{}", config.spi_num);
 
+    let gpio_config_fn = |pins: &[PinConfig], name| {
+        let name = quote::format_ident!("{name}");
+        quote::quote! {
+            fn #name() {
+                use drv_lpc55_gpio_api::*;
+                let iocon_base = lpc55_pac::IOCON::ptr() as *const u32 as u32;
+
+                #(
+                    let (pin, conf) = drv_lpc55_gpio_api::Pins::iocon_conf_val(#pins);
+                    let base = iocon_base + 4 * pin;
+                    // SAFETY: we're relying on the enum value of the pin for
+                    // correctness here. The LPC55 IOCON Rust API has individual
+                    // functions for each pin which we aren't easy to use with a
+                    // task based setup but could actually be used here. It's
+                    // not clear how much benefit we'd actually get from that
+                    // though.
+                    unsafe {
+                        core::ptr::write_volatile(base as *mut u32, conf);
+                    }
+                )*
+            }
+        }
+    };
+
+    let switch_io_out = gpio_config_fn(&config.out_cfg, "switch_io_out");
```

Comment:
(Thought: these could be a loop, e.g. over an array of `[(&config.out_cfg, "switch_io_out"), (&config_in.cfg, "switch_io_in"), ...]`)